feat(accessibility): add WCAG 2.0 AA compliant color palette

austin94 opened the pull request:
<!-- structure the Title above as the first line of a
     https://conventionalcommits.org/ message. example: "feat(buttons):
     add a muted button component". the title informs the semantic
     version bump if this PR is merged. -->

* [x] **BREAKING CHANGE?** <!-- if so, indicate why under description -->

## Description

This PR upgrades our components to the accessible blue palette.

* Updates Avatars to remove the old `isIn` and `isOut` style of props.
* Updates Tooltips to include the new `Title` and `Paragraph` components
  * Also saw that we didn't have the Garden `data` attributes so I added that as well
* Update all other packages to the new color styling
* Update all necessary tests

This is a **breaking change** due to the large visual changes (and Avatar/Tooltip changes)

## Detail

Pre-published at: http://garden.zendesk.com/react-components

## Checklist

* [x] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
* [x] :nail_care: view component styling is based on a Garden CSS
  component
* [x] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
* [x] :arrow_left: renders as expected with reversed (RTL) direction
* [x] :guardsman: includes new unit and snapshot tests
* [x] :ledger: any new files are included in the packages `src/index.js` export
* [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

jzempel commented on packages/tooltips/src/views/LightTooltip.example.md:
```diff
@@ -2,21 +2,22 @@
+      <p>
+        Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia
+        consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt.
+      </p>
```
Shouldn't be using any `<p>` tags here. Each paragraph of text belongs directly in a `<Paragraph>` element in order to receive intended margining.

austin94 replied:
Fixed.

austin94 commented:
Updated all color usages. Also found that `eslint` was missing the utils folder because we weren't wrapping the globs with `"` specifically. These are now added.

jzempel commented on package.json:
```diff
@@ -13,7 +13,7 @@
     "lint": "yarn lint:css && yarn lint:js && yarn lint:markdown",
     "lint:css": "stylelint packages/*/src/**/*.js && stylelint utils/**/*.js",
-    "lint:js": "eslint packages/*/src/ utils/**/*.js --max-warnings 0",
+    "lint:js": "eslint \"packages/*/src/\" \"utils/**/*.js\" --max-warnings 0",
```
I'd prefer the simpler `eslint packages/*/src/ utils/ ...` here. Easier to read.
